From: "Myah Caron" <qsniyg@mail.com>
Subject: [PATCH v2 1/3] kernel32: Add stub for GetFirmwareEnvironmentVariableEx
Message-Id: <trinity-5a6097ab-3732-487a-9529-b04567d2b920-1584276887407@3c-app-mailcom-lxa12>
Date: Sun, 15 Mar 2020 13:54:47 +0100

Signed-off-by "Myah Caron" <qsniyg@mail.com>
---
v2: Fix offsets (the old patch was built against staging - my bad!)
Also includes two other stubs (in patches 2 and 3) that are closely related, needed for a hardware information gathering program

 dlls/kernel32/kernel32.spec |  2 ++
 dlls/kernel32/process.c     | 21 +++++++++++++++++++++
 2 files changed, 23 insertions(+)

diff --git a/dlls/kernel32/kernel32.spec b/dlls/kernel32/kernel32.spec
index ba4e89991a5..b843a612461 100644
--- a/dlls/kernel32/kernel32.spec
+++ b/dlls/kernel32/kernel32.spec
@@ -690,6 +690,8 @@
 @ stdcall -import GetFinalPathNameByHandleW(long ptr long long)
 @ stdcall GetFirmwareEnvironmentVariableA(str str ptr long)
 @ stdcall GetFirmwareEnvironmentVariableW(wstr wstr ptr long)
+@ stdcall GetFirmwareEnvironmentVariableExA(str str ptr long ptr)
+@ stdcall GetFirmwareEnvironmentVariableExW(wstr wstr ptr long ptr)
 @ stdcall -import GetFullPathNameA(str long ptr ptr)
 # @ stub GetFullPathNameTransactedA
 # @ stub GetFullPathNameTransactedW
diff --git a/dlls/kernel32/process.c b/dlls/kernel32/process.c
index 4e981f738ce..7dfa800fee9 100644
--- a/dlls/kernel32/process.c
+++ b/dlls/kernel32/process.c
@@ -1014,3 +1014,24 @@ BOOL WINAPI UmsThreadYield(void *param)
     SetLastError( ERROR_CALL_NOT_IMPLEMENTED );
     return FALSE;
 }
+
+/***********************************************************************
+ *              GetFirmwareEnvironmentVariableExA         (KERNEL32.@)
+ */
+DWORD WINAPI GetFirmwareEnvironmentVariableExA(LPCSTR name, LPCSTR guid, PVOID buffer, DWORD size, PDWORD attributes)
+{
+    FIXME("stub: %s %s %p %u %p\n", debugstr_a(name), debugstr_a(guid), buffer, size, attributes);
+    SetLastError(ERROR_INVALID_FUNCTION);
+    return 0;
+}
+
+
+/***********************************************************************
+ *              GetFirmwareEnvironmentVariableExW         (KERNEL32.@)
+ */
+DWORD WINAPI GetFirmwareEnvironmentVariableExW(LPCWSTR name, LPCWSTR guid, PVOID buffer, DWORD size, PDWORD attributes)
+{
+    FIXME("stub: %s %s %p %u %p\n", debugstr_w(name), debugstr_w(guid), buffer, size, attributes);
+    SetLastError(ERROR_INVALID_FUNCTION);
+    return 0;
+}
